packages/widget_toolkit_otp/README.md

```diff
@@ -0,0 +1,105 @@
+this behaviour, you can set the preferDoubleDigitsForTime parameter to `false`.
+
+### AutomatedResendCodeButton
+
+AutomatedResendCodeButton maintains its own state. Once the button is being pressed, it goes through 
+a few statuses (loading, codeSent, disabled, active, error). The button is enabled only in active state. 
+There are a lot of options to be customised as providing icons, labels, colorStyles, textStyle e.t. 
+To prevent user from spam pressing the button it gets disabled for long after the onPressed callback 
+is executed. That way we are not going to sent too much codes and confuse the user. Uses the CountdownWidget under the hood.
+
+### ResendCodeButton
+
+ResendCodeButton might be maintained from your business logic, but by default it uses SmsCodeBloc. 
+It builds TextButton with great amount of customization options and authomated styling.
+
+### SmsPhoneNumberField
+
+SmsPhoneNumberField is kind of builder witch provide phone number and callback to update saved phone
+number. It gets handy to build (if needed) featured field, wrapped int SmsCodeWidget.
+
+### ResendButtonTimer
+
+ResendButtonTimer wraps CountdownWidget with the business logic, provided from SmsCodeBloc. It have 
+builder, to be totally recreated, using the same logic.
+
+### ValidityWidget
+
+ValidityWidget displays how long will be valid lately sent to the user code. This widget can not be 
+used outside the SmsCodeBloc and its content might be fully recreated using its builder.
+
+### SmsCodeWidget
+
+SmsCodeWidget is user friendly way to provide all dependencies for the implemented business logic and,
+in the same step, arrange the page content depending on bloc's states and events.
+
+## How to use
+
+In order to start using this package you need to add the dependency to the `widget_toolkit_otp` in your `pubspec.yaml` file.
+
+```yaml
+widget_toolkit_otp:
+  git:
+    url: https://github.com/Prime-Holding/widget_toolkit/tree/develop/
+    path: packages/widget_toolkit_otp
+widget_toolkit:
+  git:
+    url: https://github.com/Prime-Holding/widget_toolkit/tree/develop/
+    path: packages/widget_toolkit
+```
+
+Afterwards, you can import the package by including the following line:
+
+`import 'package:widget_toolkit_otp/widget_toolkit_otp.dart';`
+
+You should also import the widget_toolkit package:
+
+     import 'package:widget_toolkit/widget_toolkit.dart';
+
+after which you are ready to start using the widgets in your app.
+
+There are two ways of using this package - to use prebuilt widgets to visualize your own business 
+logic or to use the provided SmsCodeBloc and arrange widgets in a page, however you want.
+Widgets that you can use without SmsCodeBloc are CountdownWidget, AutomatedResendCodeButton, 
+SmsCodeField(set useInternalCommunication = false) and ResendCodeButton(set useInternalCommunication = false).
+
+To use the implemented business logic, you first need to provide in the context SmsCodeBloc and 
```

Comment:
Please add some code snippets similar to this one https://pub.dev/documentation/widget_toolkit/latest/topics/Item%20Picker-topic.html